packages/react/src/components/Combobox/Combobox.tsx

```diff
@@ -319,42 +447,115 @@ const Combobox = forwardRef<HTMLDivElement, ComboboxProps>(
         onChange?.(event);
         // istanbul ignore else
         if (!isControlled) {
-          setValue(event.target.value);
+          setInputValue(event.target.value);
         }
       },
       [isControlled, onChange]
     );
 
-    const handleSelectionChange = useCallback(
-      ({
-        target,
-        value: listboxValue,
-        previousValue
-      }: Parameters<
-        Exclude<
-          React.ComponentProps<typeof Listbox>['onSelectionChange'],
-          undefined
-        >
-      >[0]) => {
-        const stringValue =
-          listboxValue?.toString() || /* istanbul ignore next */ '';
-
-        // istanbul ignore else
-        if (!isControlled) {
-          setValue(stringValue);
-        }
+    const handleRemove = useCallback(
```

Comment:
Nit: Handle remove of what? We could benefit from a more specific name, something like:

```suggestion
    const handleRemovePill = useCallback(
```